File: backend/app/schemas/shop_info.py
from pydantic import BaseModel

from app.schemas.base import ORMBase


class ShopInfoOut(ORMBase):
    shop_name: str
    address: str
    phone: str
    whatsapp: str | None
    email: str | None
    instagram_url: str | None
    opening_hours: str
    latitude: float | None
    longitude: float | None


class ShopInfoUpdate(BaseModel):
    shop_name: str
    address: str
    phone: str
    whatsapp: str | None = None
    email: str | None = None
    instagram_url: str | None = None
    opening_hours: str = ""
    latitude: float | None = None
    longitude: float | None = None

File: src/api/shopInfo.ts
import { apiRequest } from './client'
import type { ShopInfo } from '../types'

export interface ShopInfoInput {
  shop_name: string
  address: string
  phone: string
  whatsapp?: string | null
  email?: string | null
  instagram_url?: string | null
  opening_hours: string
  latitude?: number | null
  longitude?: number | null
}

export function getShopInfo(): Promise<ShopInfo> {
  return apiRequest<ShopInfo>('/shop-info')
}

export function updateShopInfo(input: ShopInfoInput): Promise<ShopInfo> {
  return apiRequest<ShopInfo>('/shop-info', { method: 'PATCH', body: input })
}
